### PR TITLE
Sync master/1.1.5 into release/1.2 [full merge, no squash]

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Update release branch with latest commits from master.
Includes bug-fix release 1.1.5

Continuation of #5483

### CODE REVIEW

**Comment by Borda** [comment]
@carmocca mind fix the failing test?
```
    def on_train_start(self, trainer, pl_module):
>       self.log("on_train_start", 1)
E       AttributeError: 'LoggingCallback' object has no attribute 'log'

tests/trainer/logging_process/test_train_loop_logging_1_0.py:826: AttributeError
```

**Review by Borda — CHANGES_REQUESTED** [review]
@carmocca mind fix the failing test?
```
    def on_train_start(self, trainer, pl_module):
>       self.log("on_train_start", 1)
E       AttributeError: 'LoggingCallback' object has no attribute 'log'

tests/trainer/logging_process/test_train_loop_logging_1_0.py:826: AttributeError
```

**Comment by Borda** [comment]
@carmocca any update here?

**Comment by Borda** [comment]
pls, do not merge `release/1.2-dev` to this branch, always rebase :]

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
when approximately is this expected to be merged into dev branch?

**Comment by Borda** [comment]
> when approximately is this expected to be merged into dev branch?

I hope just after @carmocca resolves conflicts :]

File: CHANGELOG.md [hunk 1]
```diff
@@ -69,9 +68,6 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added compositional metrics ([#5464](https://github.com/PyTorchLightning/pytorch-lightning/pull/5464))
 
 
-- Added Trainer method `predict(...)` for high performence predictions ([#5579](https://github.com/PyTorchLightning/pytorch-lightning/pull/5579))
```
Comment (Borda) [inline CHANGELOG.md:72]: why was this removed? this section shall not be touched by sync...

**Comment by Borda** [comment]
@carmocca @SeanNaren  mind double-check the sync https://github.com/PyTorchLightning/pytorch-lightning/pull/5583#discussion_r565735204

**Comment by Borda** [comment]
@carmocca can we resolve conflicts and be sure we do not change anything in new accelerators/plugins since the master version in legacy, but git may not recognize it correctly...

**Review by Borda — CHANGES_REQUESTED** [review]
seems that in some other PRs `test_mode` was removed but it was not well propagated...
`E       NameError: name 'test_mode' is not defined`

**Comment by carmocca** [comment]
#5388 should not have landed to master and we have tricky conflicts between that PR and the accelerator refactor.

@justusschock @rohitgr7 (conflict authors) Can you fix this?

**Comment by awaelchli** [comment]
@carmocca regarding accelerator refactor, what kind of conflicts are you experiencing?
all current logic has moved to the .legacy subfolders, so all bugfixes for accelerators should target these files under .legacy.

**Comment by carmocca** [comment]
As I mentioned, #5388 (a refactor PR) landed to master (shouldn't have). This means that we have two diverging refactors. #5388 from master and the accelerator one from release.

This is why I'd prefer to have one of the authors fix these conflicts.

**Comment by Borda** [comment]
there is the last failing test...
`FAILED tests/trainer/test_trainer.py::test_trainer_predict_ddp_cpu`

**Comment by awaelchli** [comment]
In the error message you can see:

```
E           results = self.train_or_test()
E         File "/home/runner/work/pytorch-lightning/pytorch-lightning/pytorch_lightning/accelerators/legacy/accelerator.py", line 73, in train_or_test
E           results = self.trainer.train()
E         File "/home/runner/work/pytorch-lightning/pytorch-lightning/pytorch_lightning/trainer/trainer.py", line 596, in train
E           self.train_loop.run_training_epoch()
E         File "/home/runner/work/pytorch-lightning/pytorch-lightning/pytorch_lightning/trainer/training_loop.py", line 535, in run_training_epoch
E           batch_output = self.run_training_batch(batch, batch_idx, dataloader_idx)
```

It's running a training batch instead of predicting, so there must be something wrong with the routing of predict based on the `trainer.running_stage` or `trainer._predicting` attributes.

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

**Comment by Borda** [comment]
@carmocca come new conflicts :/

File: pytorch_lightning/trainer/training_loop.py [hunk 2]
```diff
@@ -141,6 +141,39 @@ def setup_training(self):
...
+
+        # wait for all to join if on distributed
+        self.trainer.accelerator_backend.barrier("setup_training")
+
+        # register auto-resubmit when on SLURM
+        self.trainer.slurm_connector.register_slurm_signal_handlers()
+
+        # --------------------------
+        # Pre-train
+        # --------------------------
```
Comment (awaelchli) [inline pytorch_lightning/trainer/training_loop.py]: I think this was incorrectly merged here in the last commit. Shouldn't be here right @rohitgr7 ?

Comment (rohitgr7) [reply pytorch_lightning/trainer/training_loop.py]: yes shouldn't be here, should be in `setup_trainer`. will update in few hours.

**Review by Borda — CHANGES_REQUESTED** [review]
see https://github.com/PyTorchLightning/pytorch-lightning/pull/5583#discussion_r569363367

File: pytorch_lightning/trainer/training_loop.py (hunk 2 above)
Comment (tchaton) [reply pytorch_lightning/trainer/training_loop.py]: Removed it.

**Comment by rohitgr7** [comment]
pls don't squash the commits right now. need to check a few things.

**Comment by Borda** [comment]
@tchaton we shall not merge 1.2 here as we want to have it berged without squash on top of 1.2
> Merge branch 'release/1.2-dev' into sync-1.1.5

**Comment by rohitgr7** [comment]
not sure why isort is behaving this way:
```
-from pytorch_lightning import seed_everything, Trainer
+from pytorch_lightning import seed_everything
+from pytorch_lightning import Trainer
-from pytorch_lightning.utilities import _FAIRSCALE_AVAILABLE, _NATIVE_AMP_AVAILABLE
+from pytorch_lightning.utilities import _FAIRSCALE_AVAILABLE
+from pytorch_lightning.utilities import _NATIVE_AMP_AVAILABLE
-from tests.base.boring_model import BoringModel, RandomDataset
+from tests.base.boring_model import BoringModel
+from tests.base.boring_model import RandomDataset
```
is the config changed?

**Comment by carmocca** [comment]
> is the config changed?

yes: https://github.com/PyTorchLightning/pytorch-lightning/pull/5739#discussion_r568997158

**Comment by Borda** [comment]
> > is the config changed?
> 
> yes: [#5739 (comment)](https://github.com/PyTorchLightning/pytorch-lightning/pull/5739#discussion_r568997158)

yes, changing it back in #5768

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]
